types/office-js-preview/index.d.ts

```diff
@@ -11454,6 +11454,50 @@ declare namespace Office {
          *                   which is an `Office.AsyncResult` object. The EWS appointment ID is returned in the `asyncResult.value` property.
          */
         saveAsync(callback: (asyncResult: Office.AsyncResult<string>) => void): void;
+        /**
+         * Sends the appointment being composed.
+         *
+         * @remarks
+         * [Api set: Mailbox preview]
+         *
+         * **{@link https://learn.microsoft.com/office/dev/add-ins/outlook/understanding-outlook-add-in-permissions | Minimum permission level}**: **read/write mailbox**
+         *
+         * **{@link https://learn.microsoft.com/office/dev/add-ins/outlook/outlook-add-ins-overview#extension-points | Applicable Outlook mode}**: Appointment Organizer
+         *
+         * **Important**:
+         *
+         * - The `sendAsync` method is only supported in task pane and function command implementations. It isn't supported in
+         * {@link https://learn.microsoft.com/office/dev/add-ins/outlook/autolaunch | event-based handlers} or
+         * {@link https://learn.microsoft.com/office/dev/add-ins/outlook/item-multi-select | item multi-select scenarios}.
+         *
+         * @param options - An object literal that contains the `asyncContext` property. Use the `asyncContext` property to specify any object you want to access in the
+         *                  callback function.
+         * @param callback - Optional. When the method completes, the function passed in the `callback` parameter is called with a single parameter, `asyncResult`. The `asyncResult`
+         *                   parameter is an `Office.AsyncResult` object. The callback function only runs if `asyncResult.status` is `Office.AsyncResultStatus.Failed`.
```

Comment:
This is actually not a guarantee (that only the failed callback runs). We should advise them not to run any code on success of this API because the item has already been sent and so is not guaranteed to run.

Reply:
Updated based on the recent changes to the feature (here and in other `sendAsync` instances). Let me know if these notes correctly align with expected behavior.